src/console/Normalize/Processor.cs

```diff
@@ -58,7 +58,10 @@ public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
         {
             EnsureArg.IsNotNull(events);
 
-            await _retryPolicy.ExecuteAsync(async () => await ConsumeAsyncImpl(events, _templateManager.GetTemplateAsString(_templateDefinition)));
+            if (events.Any())
```

Comment:
>Any [](http://example.com/codeflow?start=23&length=3)

We have multiple enumerations over the same IEnumerable.  Would be better if we could avoid this.  Can we do an events.FirstOrDefault() in the Impl method?  If null you can return.  If you have a value then you can get your partition id.